draft-ietf-httpbis-origin-frame.md

```diff
@@ -147,16 +159,16 @@ See {{algo}} for an illustrative algorithm for processing ORIGIN frames.
 The set of origins (as per {{!RFC6454}}) that a given connection might be used for is known in this
 specification as the Origin Set.
 
-By default, the Origin Set for a connection is uninitialised. When an ORIGIN frame is first received
-and successfully processed by a client, the connection's Origin Set is defined to contain an initial
-origin.  The initial origin is composed from:
+By default, the Origin Set for a connection is uninitialised. When the ORIGIN setting ({{setting}})
+is received by a client, the connection's Origin Set is defined to contain an initial origin. The
+initial origin is composed from:
 
   - Scheme: "https"
   - Host: the value sent in Server Name Indication (SNI, {{!RFC6066}}, Section 3), converted to lower case
   - Port: the remote port of the connection (i.e., the server's port)
```

Comment:
I continue to be a little sad about the port's interaction with Alt-Svc.  At this point, we're saying that the client opened the connection to speak to a particular origin (example.com:443) via an Alternative at a different port (alt.example.com:8443).  So upon receipt of the server's settings, the connection will be valid for requests bound to https://example.com:8443, which is no outstanding request on the client, and the client can't legitimately make requests until it receives the ORIGIN frame.

If https://example.com:443 and https://example.com:8443 Alt-Svc to alt.example.com:8443, then the server can't know which origin the client thinks it's talking to except by the requests it makes.  I still think I'd initialize this to the origin (including port) which prompted opening the connection, rather than the actual server port.  The server doesn't necessarily need to know this value before the client makes a request.